Source/KerbalConstructionTime/Utilities/Utilities.cs

```diff
@@ -2122,6 +2126,131 @@ public static void CleanupDebris(string launchSiteName)
                 }
             }
         }
+
+        public static Tuple<float, List<string>> GetVesselUnlockInfo(ShipConstruct ship)
+        {
+            if (HighLogic.CurrentGame.Mode != Game.Modes.CAREER) return new Tuple<float, List<string>>(0, new List<string>());
+
+            // filter the ship parts list to those parts that are not already purchased
+            var purchasableParts = GetPartsWithPurchasability(ship.Parts).Where(kvp => kvp.Value.Status == PurchasabilityStatus.Purchasable || kvp.Value.Status == PurchasabilityStatus.Unavailable);
+            List<string> ecmPartsList = new List<string>();
+            float runningCost = 0;
+
+            // compare the part specified entry cost to the ECM database
+            foreach (var part in purchasableParts.Select(kvp => kvp.Key))
+            {
+                var rawCost = part.entryCost;
+                var ecmEstCost = RealFuels.EntryCostManager.Instance.ConfigEntryCost(part.name);
+                if (rawCost == ecmEstCost)
+                {
+                    // this part is managed by the ECM, save its name later for a batch request
+                    var sanitizedName = RealFuels.Utilities.SanitizeName(part.name);
+                    if (!ecmPartsList.Contains(sanitizedName)) 
+                        ecmPartsList.Add(sanitizedName);
+                }
+                else
+                {
+                    // this part is not in the ECM, take the raw cost
+                    runningCost += rawCost;
+                }
+            }
+
+            // filter down further to those parts that can't be unlocked with our current tech and get the tech names needed
+            var lockedParts = purchasableParts.Where(kvp => kvp.Value.Status == PurchasabilityStatus.Unavailable).Select(kvp => kvp.Key).ToList();
+            List<string> pendingTech = new List<string>();
+
+            foreach (AvailablePart p in lockedParts)
```

Comment:
Could simplify this entire loop to `lockedParts.Select(p => p.TechRequired).ToHashSet()`